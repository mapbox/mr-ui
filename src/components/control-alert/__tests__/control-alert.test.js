import { shallow } from 'enzyme';
import { testCases } from './control-alert-test-cases';
import React from 'react';

describe('ControlAlert', () => {
  describe(testCases.basicError.description, () => {
    const wrapper = shallow(
      React.createElement(
        testCases.basicError.component,
        testCases.basicError.props
      )
    );

    test('renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe(testCases.basicLocked.description, () => {
    const wrapper = shallow(
      React.createElement(
        testCases.basicLocked.component,
        testCases.basicLocked.props
      )
    );

    test('renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe(testCases.basicSuccess.description, () => {
    const wrapper = shallow(
      React.createElement(
        testCases.basicSuccess.component,
        testCases.basicSuccess.props
      )
    );

    test('renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe(testCases.dismissiveWarning.description, () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(
        React.createElement(
          testCases.dismissiveWarning.component,
          testCases.dismissiveWarning.props
        )
      );
    });

    test('renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('onButtonClick is called', () => {
      wrapper
        .find('button')
        .first()
        .props()
        .onClick();
      wrapper.update();
      expect(wrapper).toMatchSnapshot();
      expect(
        testCases.dismissiveWarning.props.onButtonClick
      ).toHaveBeenCalledTimes(1);
    });
  });

  describe(testCases.dismissiveMultilineError.description, () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(
        React.createElement(
          testCases.dismissiveMultilineError.component,
          testCases.dismissiveMultilineError.props
        )
      );
    });

    test('renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('onButtonClick is called', () => {
      wrapper
        .find('button')
        .first()
        .props()
        .onClick();
      wrapper.update();
      expect(wrapper).toMatchSnapshot();
      expect(
        testCases.dismissiveMultilineError.props.onButtonClick
      ).toHaveBeenCalledTimes(1);
    });
  });
});
