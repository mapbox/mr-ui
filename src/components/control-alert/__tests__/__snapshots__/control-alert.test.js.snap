// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlAlert basic error renders as expected 1`] = `
<div
  className="round px24 py24 flex-parent bg-red-faint"
>
  <div
    className="flex-child flex-child--no-shrink"
  >
    <Icon
      inline={true}
      name="alert"
      passthroughProps={Object {}}
      size="1.3em"
    />
  </div>
  <div
    className="flex-child flex-child--grow"
    role={null}
    tabIndex={null}
  >
    <span
      className="ml6"
    >
      Drats.
    </span>
  </div>
</div>
`;

exports[`ControlAlert basic locked renders as expected 1`] = `
<div
  className="round px24 py24 flex-parent bg-yellow-light"
>
  <div
    className="flex-child flex-child--no-shrink"
  >
    <Icon
      inline={true}
      name="lock"
      passthroughProps={Object {}}
      size="1.3em"
    />
  </div>
  <div
    className="flex-child flex-child--grow"
    role={null}
    tabIndex={null}
  >
    <span
      className="ml6"
    >
      Nope.
    </span>
  </div>
</div>
`;

exports[`ControlAlert basic success renders as expected 1`] = `
<div
  className="round px24 py24 flex-parent bg-green-light"
>
  <div
    className="flex-child flex-child--no-shrink"
  >
    <Icon
      inline={true}
      name="check"
      passthroughProps={Object {}}
      size="1.3em"
    />
  </div>
  <div
    className="flex-child flex-child--grow"
    role={null}
    tabIndex={null}
  >
    <span
      className="ml6"
    >
      Woohoo!
    </span>
  </div>
</div>
`;

exports[`ControlAlert dismissive multiline error onButtonClick is called 1`] = `
<div
  className="round px24 py24 flex-parent bg-red-faint"
>
  <div
    className="flex-child flex-child--no-shrink"
  >
    <Icon
      inline={true}
      name="alert"
      passthroughProps={Object {}}
      size="1.3em"
    />
  </div>
  <div
    className="flex-child flex-child--grow"
    role={null}
    tabIndex={null}
  >
    <p
      className="mx6"
    >
      This is the song that never ends.
       
      <a
        className="link"
        href="http://www.mapbox.com"
      >
        Yes, it just goes on and on my friends.
      </a>
       
      Some people started singing it, not knowing what it was. And they'll continue singing it forever just because. This is the song that never ends.
    </p>
  </div>
  <Tooltip
    alignment="center"
    block={true}
    coloring="light"
    content="Dismiss"
    disabled={false}
    maxWidth="medium"
    padding="small"
    placement="top"
    respondsToClick={false}
    textSize="s"
  >
    <button
      className="bg-transparent color-gray-dark color-blue-on-hover px0 py0"
      data-test="alert-dismiss"
      onClick={[Function]}
    >
      <Icon
        inline={true}
        name="close"
        passthroughProps={Object {}}
        size="1.3em"
      />
    </button>
  </Tooltip>
</div>
`;

exports[`ControlAlert dismissive multiline error renders as expected 1`] = `
<div
  className="round px24 py24 flex-parent bg-red-faint"
>
  <div
    className="flex-child flex-child--no-shrink"
  >
    <Icon
      inline={true}
      name="alert"
      passthroughProps={Object {}}
      size="1.3em"
    />
  </div>
  <div
    className="flex-child flex-child--grow"
    role={null}
    tabIndex={null}
  >
    <p
      className="mx6"
    >
      This is the song that never ends.
       
      <a
        className="link"
        href="http://www.mapbox.com"
      >
        Yes, it just goes on and on my friends.
      </a>
       
      Some people started singing it, not knowing what it was. And they'll continue singing it forever just because. This is the song that never ends.
    </p>
  </div>
  <Tooltip
    alignment="center"
    block={true}
    coloring="light"
    content="Dismiss"
    disabled={false}
    maxWidth="medium"
    padding="small"
    placement="top"
    respondsToClick={false}
    textSize="s"
  >
    <button
      className="bg-transparent color-gray-dark color-blue-on-hover px0 py0"
      data-test="alert-dismiss"
      onClick={[Function]}
    >
      <Icon
        inline={true}
        name="close"
        passthroughProps={Object {}}
        size="1.3em"
      />
    </button>
  </Tooltip>
</div>
`;

exports[`ControlAlert dismissive warning onButtonClick is called 1`] = `
<div
  className="round px24 py24 flex-parent bg-yellow-light"
>
  <div
    className="flex-child flex-child--no-shrink"
  >
    <Icon
      inline={true}
      name="alert"
      passthroughProps={Object {}}
      size="1.3em"
    />
  </div>
  <div
    className="flex-child flex-child--grow"
    role={null}
    tabIndex={null}
  >
    <span
      className="mx6"
    >
      Uh oh...
    </span>
  </div>
  <Tooltip
    alignment="center"
    block={true}
    coloring="light"
    content="Dismiss"
    disabled={false}
    maxWidth="medium"
    padding="small"
    placement="top"
    respondsToClick={false}
    textSize="s"
  >
    <button
      className="bg-transparent color-gray-dark color-blue-on-hover px0 py0"
      data-test="alert-dismiss"
      onClick={[Function]}
    >
      <Icon
        inline={true}
        name="close"
        passthroughProps={Object {}}
        size="1.3em"
      />
    </button>
  </Tooltip>
</div>
`;

exports[`ControlAlert dismissive warning renders as expected 1`] = `
<div
  className="round px24 py24 flex-parent bg-yellow-light"
>
  <div
    className="flex-child flex-child--no-shrink"
  >
    <Icon
      inline={true}
      name="alert"
      passthroughProps={Object {}}
      size="1.3em"
    />
  </div>
  <div
    className="flex-child flex-child--grow"
    role={null}
    tabIndex={null}
  >
    <span
      className="mx6"
    >
      Uh oh...
    </span>
  </div>
  <Tooltip
    alignment="center"
    block={true}
    coloring="light"
    content="Dismiss"
    disabled={false}
    maxWidth="medium"
    padding="small"
    placement="top"
    respondsToClick={false}
    textSize="s"
  >
    <button
      className="bg-transparent color-gray-dark color-blue-on-hover px0 py0"
      data-test="alert-dismiss"
      onClick={[Function]}
    >
      <Icon
        inline={true}
        name="close"
        passthroughProps={Object {}}
        size="1.3em"
      />
    </button>
  </Tooltip>
</div>
`;
